mempool/src/pool/store/mod.rs

```diff
@@ -74,7 +74,7 @@ type StrictDropPolicy = mem_usage::NoOpDropPolicy;
 
 type TrackedMap<K, V> = Tracked<BTreeMap<K, V>>;
 type TrackedSet<K> = Tracked<BTreeSet<K>>;
-type TrackedTxIdMultiMap<K> = TrackedMap<K, TrackedSet<Id<Transaction>>>;
+type TrackedTxIdMultiMap<K> = TrackedSet<(K, Id<Transaction>)>;
```

Comment:
This is the change. Everything else is to make the code work with this representation, usually simplifying the code in the process.